pkg/cmd/roachtest/roachtestutil/mixedversion/README.md

```diff
@@ -311,70 +314,221 @@ mvt.InMixedVersion(
+
+``` go
+mixedversion.NeverUseFixtures
+```
+
+This option tells the test planner to never use cockroachdb [fixtures](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/roachtest/fixtures) when setting up the cluster. These fixtures are useful as they have been maintained by upgrading a cockroachdb cluster all the way back since v1.0, and they sometimes expose bugs. However, a key limitation of the fixtures is that they only work in 4-node clusters. If the test uses a differently sized cluster, this option is necessary.
+
+``` go
+mixedversion.MaxUpgrades(2)
+```
+
+This option instructs the planner to generate test plans that run no more than the given amount of version upgrades. Useful to limit the test's running time, as upgrades can take a long time to run if the test has long running user hooks. See also: `mixedversion.MinUpgrades()` and `mixedversion.NumUpgrades()`.
+
+``` go
+mixedversion.MinimumSupportedVersion("v24.1.0")
+```
+
+`mixedversion` tests may perform [several upgrades](#high-level-overview) before finalizing the upgrade to the current version. If the feature being tested was introduced in a specific, supported version then the test should use this option to indicate that. Note that this **does not** mean that the cluster will bootstrapped in this version. It could be created in an older version and upgraded to the minimum supported version. User hooks are only called once the cluster reaches the minimum supported version.
+
+``` go
+mixedversion.AlwaysUseLatestPredecessors
+```
+
+By default, `mixedversion` tests will use random predecessors when upgrading the cluster to the current version. Sometimes, however, this can lead to flakes if the test exposes a bug already fixed in a patch release. By using this option, the test planner will always pick the latest available release for each release series.
+
+``` go
+mixedversion.EnabledDeploymentModes(mixedversion.SystemOnly)
+```
+
+Sometimes, a certain test is incompatible (or flakes) with a particular [deployment mode](#deployment-modes). This option allows the test author to specify the deployments modes compatible with a test.
+
+``` go
+mixedversion.DisableMutators(mixedversion.PreserveDowngradeOptionRandomizerMutator)
+```
+
+Use this option to disable specific [mutators](#mutators) that are incompatible with the test.
+
+### Deployment Modes
+
+By default, each run of a `mixedversion` test happens in one of 3 possible _deployment modes_: `system-only`, `shared-process`, and `separate-process`. In the latter two options, the framework will create a test tenant, and tests should exercise the feature they are testing by invoking it on the tenant.
```

Comment:
Nit: might be worth mentioning that all three modes contain the "system" service, i.e., `h.System` is never null.

Reply:
Done.